test/dsets.c

```diff
@@ -15267,7 +15276,7 @@ test_h5s_plist(void)
     /* Attempt to 'OR' block with invalid dimensions into the selection */
     H5E_BEGIN_TRY
     {
-        ret = H5Pset_dataset_io_hyperslab_selection(dxpl_id_copy, 2, H5S_SELECT_OR, &start, &stride, &count,
+        ret = H5Pset_dataset_io_hyperslab_selection(dxpl_id_copy, H5S_MAX_RANK + 1, H5S_SELECT_OR, &start, &stride, &count,
```

Comment:
Specifying 2 for the dimensionality here caused an overflow since start, stride, count, block are single values, not 2-dimensional arrays. Since the test was trying to 'OR' an invalid block, use H5S_MAX_RANK + 1 for the dimensionality, which should fail early.